File: services/payment-service/src/main/java/io/minishop/payment/web/dto/PgChargeRequest.java
package io.minishop.payment.web.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record PgChargeRequest(
        @NotNull Long paymentId,
        @NotNull @PositiveOrZero BigDecimal amount
) {}

File: services/payment-service/src/main/kotlin/io/minishop/payment/web/dto/PgChargeRequest.kt
package io.minishop.payment.web.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

@JvmRecord
data class PgChargeRequest(
    @field:NotNull val paymentId: Long?,
    @field:NotNull @field:PositiveOrZero val amount: BigDecimal?,
)
